tools/shoestring/tests/internal/test_CertificateFactory.py

```diff
@@ -269,6 +269,29 @@ def test_cannot_generate_node_certificate_without_common_name(self):
 				with self.assertRaisesRegex(RuntimeError, 'Node common name cannot be empty'):
 					factory.generate_node_certificate('')
 
+	def test_can_reuse_ca_certificate(self):
+		with tempfile.TemporaryDirectory() as package_directory:
+			ca_certificate_path = Path(package_directory) / 'ca.crt.pem'
+
+			# - create a CA private key
+			with tempfile.TemporaryDirectory() as certificate_directory:
+				ca_private_key_path = self._create_ca_private_key(certificate_directory)
+
+				with CertificateFactory(self._create_executor(), ca_private_key_path) as factory:
+					# create CA certificate
+					ca_common_name = 'my CA common name'
+					factory.generate_ca_certificate(ca_common_name)
+					factory.package(package_directory)
+
+					# Act:
+					with CertificateFactory(self._create_executor(), ca_private_key_path) as renew_factory:
+						renew_factory.reuse_ca_certificate(ca_common_name, Path(package_directory))
+
+						# Assert:
+						assert Path('ca.crt.pem').exists()
+						assert Path('ca.cnf').exists()
+						assert ca_certificate_path.exists()
```

Comment:
i don't think this test is correct because i believe thes would be created by (first) CertificateFactory. to emulate real behavior, i think you want two temporary directories one for each CertificateFactory

Reply:
hmm..I will take a look but I thought the second CertificateFactory would create another temp folder? 🤔

Reply:
In which case the ``factory.reuse_ca_certificate`` would fail to create the ``new_certs`` folder since its already present.  but will move the second CertificateFactory for clarity

Reply:
Ok.  Will add a sanity check to verify that the first CertificateFactory folder was removed.

``ca_certificate_filepath`` will exist since this is where the certs were install from the first CertificateFactory

In the second CertificateFactory, the ca cert is copied from the ``ca_certificate_filepath``